test/service/test_pagination.py

```diff
@@ -62,25 +73,25 @@ def assert_page2_correct(self, json_response, json_response_second, sort_order):
         """
         num_hits_first = len(json_response['hits'])
         num_hits_second = len(json_response_second['hits'])
-        if 'search_before' in json_response_second['pagination']:
-            self.assertEqual(json_response_second['pagination']['search_before'],
-                             json.dumps(json_response_second['hits'][0]['entryId']),
-                             "search_before on second page not set to first returned document on second page, "
-                             + "order=" + sort_order)
-            self.assertIsNotNone(json_response_second['pagination']['search_before_uid'],
-                                 "No search_before_uid returned on second page")
-        else:
-            self.fail("search_before not set on second page of results, sortOrder=" + sort_order)
-
-        if 'search_after' in json_response['pagination']:
-            self.assertEqual(json_response_second['pagination']['search_after'],
-                             json.dumps(json_response_second['hits'][num_hits_second - 1]['entryId']),
-                             "search_after on second page not set to last returned document on second page, "
-                             + "order=" + sort_order)
-            self.assertIsNotNone(json_response['pagination']['search_after_uid'],
-                                 "No search_after_uid returned on second page")
-        else:
-            self.fail("search_after not set on second page of results, sortOrder=" + sort_order)
```

Comment:
We should add a `with self.subTest(sort_order=sort_order)` instead of interpolating `sort_order` into all of those assertion failure messages.